vms/proposervm/block/block.go

```diff
@@ -48,69 +67,171 @@ type statelessUnsignedBlock struct {
 	PChainHeight uint64 `serialize:"true"`
 	Certificate  []byte `serialize:"true"`
 	Block        []byte `serialize:"true"`
+	VRFSig       []byte `serialize:"true"`
+}
+
+type unsignedBlockTypes interface {
```

Comment:
See Dhruba's point above about the codec's versioning. Also, I think we could instead declare this interface as:

```
type unsignedBlockTypes interface {
	ParentID() ids.ID
	Block() []byte
	VRFSig() []byte
	PChainHeight() uint64
	innerCertificate() []byte
	innerTimestamp() int64
}
```

And then implement that interface for `statelessUnsignedBlock` and `statelessUnsignedBlockV0` so that below, we don't need to `switch` on the type and we can do, for example:

```go
func (b *statelessBlock[T]) ParentID() ids.ID {
	return b.StatelessBlock.ParentID()
}
```
without using a switch.

Reply:
the versioned codec eliminates the needs for that..